### PR TITLE
dynamic required reviewers: only track addition of patterns to acceptance-test-config

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Before this PR teams would get required for a review even if a sensitive keyword is removed from `acceptance-test-config.yml`. We only want to track addition of specific keyword, we don't care about their deletion:
* We want to know when a bypass reason is added, not when it's removed
* We want to know when disable_for_version is added, not when it's removed
* We want to knwow when test strictness level is set, not when it's removed

## How
Use `git diff --word-diff` + smart output parsing.

### CODE REVIEW

**Comment by evantahler** [comment]
Perhaps the bot went a little extra with the deletions on this PR?

File: tools/ci_connector_ops/ci_connector_ops/utils.py [hunk 1]
```diff
@@ -54,27 +54,27 @@ def get_connector_name_from_path(path):
...
 
     Returns:
         Set[Connector]: Set of connectors that were changed
     """
     airbyte_repo = git.Repo(search_parent_directories=True)
+    changed_acceptance_test_config_paths = []
+    word_diff_command_args = ("--word-diff", "--unified=0", f"-G{keyword}", DIFFED_BRANCH)
+    word_diff = airbyte_repo.git.diff(*word_diff_command_args)
+    current_file = None
+    for diff_line in word_diff.split("\n"):
```
Comment (sherifnada) [inline tools/ci_connector_ops/ci_connector_ops/utils.py:72]: Should we unit test this?

**Review by sherifnada — APPROVED** [review]

**Review by bnchrch — APPROVED** [review]
Would love to see this merged!

**Comment by evantahler** [comment]
@bnchrch - Augustin is out this week.  You can merge this if you feel comfortable managing this

**Comment by octavia-squidington-iii** [comment]
##  source-braintree test report (commit [`34a3491ad0`](https://github.com/airbytehq/airbyte/pull/26943/commits/34a3491ad0534cb75c7fe4218e53ea07b0575e13)) - ❌

⏲️  Total pipeline duration: 36.51s 

| Step                                                                    | Result   |
|:------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-braintree/metadata.yaml | ✅        |
| Connector version semver check                                          | ✅        |
| QA checks                                                               | ✅        |
| Code format checks                                                      | ✅        |
| Connector package install                                               | ✅        |
| Build source-braintree docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                              | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_dynamic-review-requirements-tweak/1689031788/34a3491ad0534cb75c7fe4218e53ea07b0575e13/source-braintree/0.1.5/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-braintree test
```

**Comment by octavia-squidington-iii** [comment]
##  source-prestashop test report (commit [`34a3491ad0`](https://github.com/airbytehq/airbyte/pull/26943/commits/34a3491ad0534cb75c7fe4218e53ea07b0575e13)) - ✅

⏲️  Total pipeline duration: 18mn16s 

| Step                                                                     | Result   |
|:-------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/source-prestashop/metadata.yaml | ✅        |
| Connector version semver check                                           | ✅        |
| QA checks                                                                | ✅        |
| Code format checks                                                       | ✅        |
| Connector package install                                                | ✅        |
| Build source-prestashop docker image for platform linux/x86_64           | ✅        |
| Unit tests                                                               | ✅        |
| Integration tests                                                        | ✅        |
| Acceptance tests                                                         | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/augustin_dynamic-review-requirements-tweak/1689031788/34a3491ad0534cb75c7fe4218e53ea07b0575e13/source-prestashop/1.0.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=source-prestashop test
```

**Pull request closed without merging**